typing/typemod.ml

```diff
@@ -1037,6 +1037,37 @@ let check_no_modal_modules ~env modes =
   | None -> ()
   | Some loc -> raise(Error(loc, env, Modal_module_not_supported))
 
+let apply_pmd_modalities env sig_modalities pmd_modalities mty =
+  let modalities =
+    match pmd_modalities with
+    | [] -> sig_modalities
+    | _ :: _ ->
+      Typemode.transl_modalities ~maturity:Alpha Immutable [] pmd_modalities
```

Comment:
Maybe leave a comment explaining this is broken, but it is broken in the same way as `val` declarations.